internal/service/influxdb/resource_influxdb_database_test.go

```diff
@@ -0,0 +1,184 @@
+package influxdb_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	acc "github.com/aiven/terraform-provider-aiven/internal/acctest"
+
+	"github.com/aiven/aiven-go-client"
+	"github.com/aiven/terraform-provider-aiven/internal/schemautil"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aiven_influxdb_database", &resource.Sweeper{
+		Name: "aiven_influxdb_database",
+		F:    acc.SweepDatabases,
+		Dependencies: []string{
+			"aiven_connection_pool",
+		},
+	})
+}
+
+func TestAccAivenInfluxDBDatabase_basic(t *testing.T) {
+	resourceName := "aiven_influxdb_database.foo"
+	rName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	rName2 := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acc.TestAccPreCheck(t) },
+		ProviderFactories: acc.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckAivenInfluxDBDatabaseResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccInfluxDBDatabaseResource(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAivenInfluxDBDatabaseAttributes("data.aiven_influxdb_database.database"),
+					resource.TestCheckResourceAttr(resourceName, "project", os.Getenv("AIVEN_PROJECT_NAME")),
+					resource.TestCheckResourceAttr(resourceName, "service_name", fmt.Sprintf("test-acc-sr-%s", rName)),
+					resource.TestCheckResourceAttr(resourceName, "database_name", fmt.Sprintf("test-acc-db-%s", rName)),
+					resource.TestCheckResourceAttr(resourceName, "termination_protection", "false"),
+				),
+			},
+			{
+				Config:                    testAccInfluxDBDatabaseTerminationProtectionResource(rName2),
+				PreventPostDestroyRefresh: true,
+				ExpectNonEmptyPlan:        true,
+				PlanOnly:                  true,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "project", os.Getenv("AIVEN_PROJECT_NAME")),
+					resource.TestCheckResourceAttr(resourceName, "service_name", fmt.Sprintf("test-acc-sr-%s", rName2)),
+					resource.TestCheckResourceAttr(resourceName, "database_name", fmt.Sprintf("test-acc-db-%s", rName2)),
+					resource.TestCheckResourceAttr(resourceName, "termination_protection", "true"),
+				),
+			},
```

Comment:
Can also have a test step for the `import` statement, since we support it as well